### PR TITLE
PWN-3853 - show no tokens text when no tokens found by checking through search

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3853

## Description of Work

show no tokens empty text when no tokens found by checking through search

## Screenshots (Optional)
Android Emulator - Pixel_5_API_30:5556 2022-06-15 13-33-53

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -337,6 +337,7 @@
     <string name="send_fees_format">Fees: %s</string>
     <string name="send_paste">Paste</string>
     <string name="send_multiple_accounts">Multiple accounts were found</string>
+    <string name="send_no_tokens">The token not found. Try to change name</string>
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/res/values/strings.xml:340]: it is in Jira ticket

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenContract.kt [hunk 2]
```diff
@@ -6,7 +6,7 @@ import org.p2p.wallet.home.model.Token
 
 interface SelectTokenContract {
     interface View : MvpView {
-        fun showTokens(items: List<Token>)
+        fun showTokens(items: List<Token>, filtered: Boolean = false)
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenContract.kt:9]: ok i will make it in 3rd way but on my own =)

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenContract.kt [hunk 3]
```diff
@@ -6,7 +6,9 @@ import org.p2p.wallet.home.model.Token
 
 interface SelectTokenContract {
     interface View : MvpView {
-        fun showTokens(items: List<Token>, filtered: Boolean = false)
+        fun showTokens(items: List<Token>)
+        fun setEmptyViewVisibility(visible: Boolean)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenContract.kt:10]: isVisible

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt [hunk 4]
```diff
@@ -72,14 +72,16 @@ class SelectTokenFragment :
...
-            messageTextView.isVisible = isEmpty && filtered
-        }
+    }
+
+    override fun setEmptyViewVisibility(visible: Boolean) {
+        binding.emptyTextView.isVisible = visible
+    }
+
+    override fun setNoSearchedTokensViewVisibility(visible: Boolean) {
+        binding.messageTextView.isVisible = visible
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt:84]: well, it's not message, it's more like `tokenNotFoundTextView`, not an ordinary message

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt [hunk 5]
```diff
@@ -72,14 +72,16 @@ class SelectTokenFragment :
...
-            emptyTextView.isVisible = isEmpty && !filtered
-            messageTextView.isVisible = isEmpty && filtered
-        }
+    }
+
+    override fun setEmptyViewVisibility(visible: Boolean) {
+        binding.emptyTextView.isVisible = visible
+    }
+
+    override fun setNoSearchedTokensViewVisibility(visible: Boolean) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt:83]: setTokenNotFoundViewVisibility

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenPresenter.kt [hunk 6]
```diff
@@ -5,23 +5,32 @@ import org.p2p.wallet.home.model.Token
...
-    BasePresenter<SelectTokenContract.View>(),
-    SelectTokenContract.Presenter {
+class SelectTokenPresenter(
+    private val tokens: List<Token>
+) : BasePresenter<SelectTokenContract.View>(), SelectTokenContract.Presenter {
 
-    override fun load() {
-        view?.showTokens(tokens)
-    }
+    override fun load() = showTokens()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenPresenter.kt:12]:
```suggestion
    override fun load() {
        showTokens()
    }
```

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenPresenter.kt [hunk 7]
```diff
@@ -5,23 +5,32 @@ import org.p2p.wallet.home.model.Token
...
 
     override fun search(searchText: String) {
         if (searchText.length < QUERY_MIN_LENGTH) {
-            view?.showTokens(tokens)
+            view?.setNoSearchedTokensViewVisibility(false)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenPresenter.kt:16]: named arg please

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenPresenter.kt [hunk 8]
```diff
@@ -5,23 +5,32 @@ import org.p2p.wallet.home.model.Token
 
 private const val QUERY_MIN_LENGTH = 2
 
-class SelectTokenPresenter(private val tokens: List<Token>) :
-    BasePresenter<SelectTokenContract.View>(),
-    SelectTokenContract.Presenter {
+class SelectTokenPresenter(
+    private val tokens: List<Token>
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenPresenter.kt:9]: allTokens to create a separation from filteredTokens?

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**